src/main/java/com/viaversion/viafabricplus/protocoltranslator/util/ProtocolVersionDetector.java

```diff
@@ -43,30 +41,27 @@ public final class ProtocolVersionDetector {
      * @return The protocol version of the server
      */
     public static ProtocolVersion get(final InetSocketAddress serverAddress, final ProtocolVersion clientVersion) {
-        final MCPingResponse response = MCPing
-                .pingModern(clientVersion.getOriginalVersion())
-                .address(serverAddress)
-                .noResolve()
-                .timeout(TIMEOUT, TIMEOUT)
-                .getSync();
-
-        if (response.version.protocol == clientVersion.getOriginalVersion()) { // If the server is on the same version as the client, we can just connect
+        // All servers (I have tested a few million...) will always respond with their highest supported version if you ping with -1
+        final JavaPinger.Response response = JavaPinger.ping(serverAddress, TIMEOUT, -1);
```

Comment:
Vanilla will never send -1 as id, this is bad and servers could use it to intentionally break auto detect

Reply:
> [If the client is pinging to determine what version to use, by convention -1 should be set.](https://minecraft.wiki/w/Minecraft_Wiki:Projects/wiki.vg_merge/Server_List_Ping)

I believe this advice comes from the Mojang developers.

Reply:
wiki.vg isn't written by Mojang developers and information on it has been wrong multiple times. I would not blindly trust that. They even mention "Setting invalid (nonexistent) version as the protocol version might cause some servers to close connection after this packet" which is exactly the concern here

Reply:
This wiki is not official, besides the vanilla client never sends -1 but always his own version so this is either something they make up or is outdated information.

Reply:
Is there any disadvantage when using the native client version?

Reply:
The version that `-1` reports on SLP is the native supported version of the server which means the Via client will be doing the version to version protocol translation rather than the server.  Good for ViaFabricPlus users, as in those instances, the user benefits from new versions of the mod.

Reply:
This should also happen with the native client version, please revert unless there is a specific reason why -1 would be better (I only see dissadvantages like the server detecting this as bad)